src/data-root.ts

```diff
@@ -0,0 +1,46 @@
+import localforage from 'localforage'
+
+import * as core from './core'
+import * as dns from './dns'
+import { CID } from './ipfs'
+import { UCAN_STORAGE_KEY } from './common'
+
+
+/**
+ * Get the CID of a user's data root.
+ */
+export async function dataRoot(username: string): Promise<CID> {
+  try {
+    // TODO: This'll be `files.${username}.fission.name` later
+    return await dns.lookupDnsLink(`${username}.fission.name`)
+  } catch(err) {
+    throw new Error("Could not locate user root in dns")
+  }
+}
+
+/**
+ * Update a user's data root.
+ */
+export const updateDataRoot = async (props: {
```

Comment:
thoughts on making the params `(cid: CID, props: {...})`, since `cid` is required?

Otherwise, I'd at least bump it up to the top of the `props` so it's obvious that it's the most important param

Reply:
I keep forgetting that the target is javascript and not typescript 😅 When using typescript, the current implementation should require you to pass { cid = ... } right? (because there's no `?` after the key)

Reply:
> I keep forgetting that the target is javascript and not typescript

yup same here haha :sweat_smile: 

> When using typescript, the current implementation should require you to pass { cid = ... } right?

yup! but I still think it might be a bit more intuitive as it's param, since most of our sdk is structured that way

Reply:
Updated 👍